docs/specifications/tests/Nameserver-TP/nameserver01.md

```diff
@@ -1,53 +1,107 @@
-## NAMESERVER01: A name server should not be a recursor
+# NAMESERVER01: A name server should not be a recursor
 
-### Test case identifier
-**NAMESERVER01** A name server should not be a recursor
+## Test case identifier
+**NAMESERVER01**
 
-### Objective
+## Objective
 
 To ensure consistency in DNS, an authoritative name server should not be
 configured to do recursive lookups. Also, open recursive resolvers are
 considered bad internet practice due to their capability of assisting in
-large scale DDoS attacks. The introduction to [RFC 5358](
-https://tools.ietf.org/html/rfc5358) elaborates on mixing recursor and
+large scale DDoS attacks. The introduction to [RFC 5358]
+elaborates on mixing recursor and
 authoritative functionality, and the issue is further elaborated by
-[D.J. Bernstein](http://cr.yp.to/djbdns/separation.html).
+[D.J. Bernstein].
 
-Section 2.5 of [RFC 2870](https://tools.ietf.org/html/rfc2870) have very
+Section 2.5 of [RFC 2870] have very
 specific requirement on disabling recursion functionality on root name
 servers.
 
-### Inputs
+## Inputs
 
-The domain name to be tested.
+* The domain name to be tested ("Child Zone").
 
-### Ordered description of steps to be taken to execute the test case
-1. Retrieve all address records for all the name servers using [Method
-   4](../Methods.md) and [Method 5](../Methods.md).
-2. A SOA query for an almost certainly nonexistent name sent to the each
-   name server IP address found in step 1, with the flag Recursion Desired 
-   (RD) set.
-3. If any answer of the queries made in step 2 contains an RCODE with
-   NXDOMAIN, this test case fails.
+## Ordered description of steps to be taken to execute the test case
 
-### Outcome(s)
+1. Create A queries for the following domain names:
+   1. xn--nameservertest.iis.se
+   2. xn--nameservertest.icann.org
+   3. xn--nameservertest.ripe.net
 
-If the response is a possible answer with the RCODE NXDOMAIN, this test
-case fails.
+2. Retrieve all address records for all the name servers for the
+   *Child Zone* using [Method4] and [Method5].
 
-### Special procedural requirements
+3. Repeat the following steps for each retrieved name server IP.
 
-None.
+   1. Send the three A queries over UDP.
+
+   2. If the name server does not respond with a to a query with a 
+      DNS response, then emit *[NO_RESPONSE]*.
+
+   3. If the response comes with the RA flag set, then 
+      emit *[IS_A_RECURSOR]*.
+
+   4. If the RCODE is NXDOMAIN in all DNS responses then 
+      emit *[IS_A_RECURSOR]*.
```

Comment:
This and previous step refer to "the response" and this step refers to the same thing as "the DNS response". Maybe they could use the same term?